onnx/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/OnnxPreTrainedModel.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.inference.onnx
+
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.Operation
+import org.jetbrains.kotlinx.dl.dataset.shape.TensorShape
+
+/**
+ * Base class for pre-trained high-level onnx models.
+ *
+ * @param [I] input type
+ * @param [R] output type
+ */
+public abstract interface OnnxPreTrainedModel<I, R> {
```

Comment:
Is _abstract_ required here?

Reply:
What do you think about renaming OnnxPreTrainedModel to something like OnnxModelHubModel? The OnnxPreTrainedModel name may confuse someone because the pre-trained model term has a broader meaning than this interface implies.

Reply:
But it's not only for the model hub, the idea for this interface to be a general purpose interface for all high-level onnx models (including user-defined). I agree that "pre-trained" here may be confusing (`OnnxInferenceModel` is also pre-trained). `OnnxHighLevelModel` perhaps? `OnnxDetector`/`OnnxDetectorModel`?

Reply:
OnnxHighLevelModel sounds better for me. OnnxDetector/OnnxDetectorModel is also misleading because one can think it's related to object detection. One more option - OnnxModelWrapper?

Reply:
`OnnxModelWrapper` feels like it's a service class name. Let's use `OnnxHighLevelModel` for now.